sygnal/sygnal.py

```diff
@@ -50,6 +50,7 @@
         "sentry": {"enabled": False},
     },
     "apps": {},
+    "database": {"name": "sqlite3", "args": {"dbfile": "sygnal.db"}},
```

Comment:
The opposite of this PR's diff was purposefully done in https://github.com/matrix-org/sygnal/pull/95/files#diff-13c51378d37f89582bc66a30f28f723b So I'm afraid this code won't work.

We either need to make `merge_left_with_defaults` more sane, or consider removing the log line as you say.

Reply:
I went for the former in https://github.com/matrix-org/sygnal/pull/111/commits/526dd8ae215c1fb238e71aab14e460deaa015a10, which probably makes more sense and is easier to do. Sorry about the force-push, I hadn't seen your review.

Reply:
I feel that rather than signalling out the database here, a better fix might just be to pull out the specific args Postgres needs rather than passing the entire config dict:

https://github.com/matrix-org/sygnal/blob/6a70885421d27fecf5145cf3d20fd0811fa37680/sygnal/sygnal.py#L139-L141

as that seems to be what the issue is here (postgres complaining about getting a `dbfile` argument?

Reply:
It is that issue indeed. But I feel like doing it that way would not win us anything and make it more painful in terms of maintenance because the current way of doing it also how we do it in all of our other projects (so creating an exception to that can easily make debugging issues more difficult - not saying we shouldn't do it, but rather that we should maintain consistency between our projects), and if an update to psycopg2 that changes the way a connection pool is configured it would require both a code change and a config change, whereas the current solution only requires the latter.

Reply:
I suppose that makes sense, although it's worth mentioning we already hard-code the options for SQLite3. Using `**args` was something introduced only recently with the postgres PRs.

To me it feels a bit weird to special-case the database here, but the comment explains its reasoning so in the grand scheme of things it's probably fine.

Reply:
Actually, testing with `ConnectionPool` it doesn't complain if you give it `dbfile` as an arg, it just ignores it.

Reply:
> Actually, testing with ConnectionPool it doesn't complain if you give it dbfile as an arg with "psycopg2", it just ignores it.

Well tests were failing with errors complaining explicitly about a `dbfile` parameter in the arguments so...

> Using `**args` was something introduced only recently with the postgres PRs.

In Sygnal yes, but surely we've been doing that in Sydent and Synapse for a while now? I definitely got that error before on Synapse due to a `path` in the args in a bad config file.